azurion/properties/Requirement1.mcf

```diff
@@ -0,0 +1,5 @@
+nu X(
+	patientOnTable: Bool = false,
+	operationOngoing: Bool = false,
+	emergencyMode: Bool = false).
+(val(!emergencyMode && operationOngoing && patientOnTable) => exists p:Pedal.[true*.pressPedal(p)]mu Y.([!switchOnXray(translatePedaltoArm(p))]Y && <true>true))
```

Comment:
Here I know what the problem is, but I'm not sure how to solve it to be honest.
I added two variables for the XRayControl component: XRayOnArm1 and XRayOnArm2. I added these so that x-rays can only be switched on if they are off, and viceversa. So, when the xray is already on, pressing the pedal will not trigger the switchOnXray action.
I think the easiest solution is for me to remove the two variables, and just go ahead and trigger the switchOnXray action for any pedal press, without checking if it is on already. Most problems in the conditions arise from this added implementation.
Also, I wouldn't even know how to set the variables for XRayControl in this modal formula.

Reply:
Also, if you initialise all of the variables as false, the ``val()`` statement will never be true, therefore it is true all the time. So variables used in the modal formula should be initialized to reflect what we want to check.

Reply:
I agree that we don't need those variables. We can only switch on the X-rays by pressing the pedal, and if the operator releases the pedal, the X-ray immediately switches off. Therefore, there is no scenario where the X-ray can be on and the pedal is released and repressed.

Reply:
You're right, that makes perfect sense. I can remove them, unless you want to take care of it.